File: UKParliament.CodeTest.Data/ViewModels/EmployeeViewModel.cs
using FluentValidation.Results;
using UKParliament.CodeTest.Data.Models;

namespace UKParliament.CodeTest.Data.ViewModels;

public class EmployeeViewModel : BaseViewModel
{
    public string? FirstName { get; set; }
    public string? LastName { get; set; }
    public string? EmployeeType { get; set; }
    public DateOnly? DoB { get; set; }

    public string? PayBand { get; set; }
    public string? Department { get; set; }
    public decimal? Salary { get; set; }
    public string? BankAccount { get; set; }

    public DateOnly? DateJoined { get; set; }
    public DateOnly? DateLeft { get; set; }

    public int? ManagerId { get; set; }
    public ShortManagerViewModel? Manager { get; set; }

    public Address? Address { get; set; }

    public bool Inactive { get; set; }
    public bool HasManager { get; set; }
    public bool IsManager { get; set; }
    public bool HasIrregularities { get; set; }
    public ValidationResult? Irregularities { get; set; }
}

File: UKParliament.CodeTest.Web/ClientApp/src/app/models/employee-view-model.ts
import { Address } from "./address";
import { BaseViewModel } from "./base-view-model";
import { ErrorBag } from "./errors/error-bag";
import { ShortManagerViewModel } from "./short-manager-view-model";

export interface EmployeeViewModel extends BaseViewModel {
  firstName?: string;
  lastName?: string;
  employeeType?: string;
  doB?: string,
  payBand?: string,
  department?: string,
  salary?: number,
  bankAccount?: string,

  dateJoined?: string,
  dateLeft?: string,

  managerId?: number | null,
  manager: ShortManagerViewModel,

  address: Address,

  inactive?: boolean,
  hasManager?: boolean,
  isManager?: boolean,
  hasIrregularities?: boolean,
  irregularities?: ErrorBag
}
